src/lib.rs

```diff
@@ -290,7 +329,23 @@ impl PrometheusMetrics {
         }
     }
 
+    #[cfg(feature = "regex-replace")]
     fn update_metrics(&self, path: &str, method: &Method, status: StatusCode, clock: SystemTime) {
+        log::trace!("About to replacing path capture groups: {}", path);
+        let mut path = String::from(path);
+        for (exp, repl) in self.replacements.clone() {
+            path = String::from(exp.replace_all(&path, &repl as &str))
+        }
```

Comment:
I'm not really happy with this as it is, because each time we're updating metrics, we copy all replacements, the path once, and then the path again for each replacement. I couldn't find a way to do this with less copies though that also makes the borrow checker happy.